sql/moz-fx-data-shared-prod/udf/monetize_search_v1/query.sql

```diff
@@ -0,0 +1,149 @@
+/*
+
+Return a boolean value indicating whether the search is monetized
+
+*/
+CREATE OR REPLACE FUNCTION udf.monetized_search(
+  engine STRING,
+  country STRING,
+  distribution_id STRING,
+  submission_date DATE
+) AS (
+  CASE
+  WHEN
+    engine LIKE 'yandex%'
+    AND country = 'UA'
+    AND submission_date < '2017-07-01'
+  THEN
+    TRUE
+  WHEN
+    (engine LIKE 'yandex%' OR engine = 'yasearch')
+    AND country IN ('RU', 'KZ', 'BY', 'TR')
+  THEN
+    TRUE
+  WHEN
+    engine IN ('bing', 'ddg', 'duckduckgo')
+  THEN
+    TRUE
+  WHEN
+    engine LIKE 'ebay%'
+    AND country IN ('US', 'GB', 'CA', 'IE', 'FR', 'IT', 'DE', 'AT', 'ES', 'NL', 'AU', 'CH')
+  THEN
+    TRUE
+  WHEN
+    engine LIKE 'amazon%'
+    AND country IN ('US', 'IT', 'GB', 'FR', 'DE', 'JP')
+  THEN
+    TRUE
+  WHEN
+    engine LIKE 'yahoo-jp%'
+    AND country IN ('JP')
+  THEN
+    TRUE
+  WHEN
+    engine = 'baidu'
+    AND country IN ('CN')
+  THEN
+    TRUE
+  WHEN
+    engine = 'ecosia'
+    AND country IN ('DE')
+  THEN
+    TRUE
+  WHEN
+    engine = 'qwant'
+    AND country IN ('FR')
+  THEN
+    TRUE
+  WHEN
+    engine = 'mailru'
+    AND country IN ('RU')
+  THEN
+    TRUE
+  WHEN
+    engine = 'seznam-cz'
+    AND country IN ('CZ')
+  THEN
+    TRUE
+  WHEN
+    engine = 'webde-suche'
+    AND country IN ('DE')
+  THEN
+    TRUE
+  WHEN
+    engine = 'rakuten'
+    AND country IN ('JP')
+  THEN
+    TRUE
+  WHEN
+    engine IN ('amazondotcom-adm', 'amazon-en-GB-adm', 'amazon-de-adm')
+  THEN
+    TRUE -- AdMarketplace
+  WHEN
+    engine LIKE('yahoo%')
+    AND submission_date < '2017-11-14'
+  THEN
+    TRUE
+  WHEN
+    engine = 'google-nocodes'
+  THEN
+    FALSE
```

Comment:
I'm not sure how much this would really matter but I would expect there's a slight performance advantage in being able to short-circuit as many clauses as possible, so putting the most common clauses as early as possible might be worth doing (in this case, that'd probably mean putting all the google clauses near the top)

Reply:
The order of condition matters for Google. If the engine is 'google-nocodes' then it's non-monetized regardless of `distribution_id`, `country`, `submission_date`. This one is mostly related to the 'canonicial' distribution coming next I think in this case.

Reply:
Yes, I mean, you could move _all_ of the google clauses to the top, since they're the most commonly used

Reply:
oh!!! I see what you mean now :)